sphinxlint/rst.py

```diff
@@ -144,6 +144,8 @@
 
 BEFORE_ROLE = r"(^|(?<=[\s(/'{\[*-]))"
 ROLE_TAG = rf":{SIMPLENAME}:"
+ROLE_TAG_ENDING_LINE_RE = re.compile(ROLE_TAG + "$")
+ROLE_TAG_STARTING_LINE_RE = re.compile("^" + ROLE_TAG)
```

Comment:
What about adding the `.search` at the end of these, and use a lower-case, function-like name?  It will make it a bit faster and avoid ugly UPPERCASE names in the middle of the functions.  Same with the one below.
If you want to avoid indirection, you could also move it in checkers.py, so they are only used there.